yarn-project/circuits.js/src/barretenberg/crypto/grumpkin/index.ts

```diff
@@ -77,19 +77,19 @@ export class Grumpkin {
    * Gets a random field element.
    * @returns Random field element.
    */
-  public getRandomFr(): Fr {
+  public getRandomFr(): GrumpkinScalar {
     this.wasm.call('ecc_grumpkin__get_random_scalar_mod_circuit_modulus', 0);
-    return Fr.fromBuffer(Buffer.from(this.wasm.getMemorySlice(0, 32)));
+    return GrumpkinScalar.fromBuffer(Buffer.from(this.wasm.getMemorySlice(0, 32)));
   }
 
   /**
    * Converts a 512 bits long buffer to a field.
    * @param uint512Buf - The buffer to convert.
    * @returns Buffer representation of the field element.
    */
-  public reduce512BufferToFr(uint512Buf: Buffer): Fr {
+  public reduce512BufferToFr(uint512Buf: Buffer): GrumpkinScalar {
     this.wasm.writeMemory(0, uint512Buf);
     this.wasm.call('ecc_grumpkin__reduce512_buffer_mod_circuit_modulus', 0, 64);
-    return Fr.fromBuffer(Buffer.from(this.wasm.getMemorySlice(64, 96)));
+    return GrumpkinScalar.fromBuffer(Buffer.from(this.wasm.getMemorySlice(64, 96)));
```

Comment:
Are we sure about this one? After all, the method is named "reduceToFr"

Reply:
Reverted in https://github.com/AztecProtocol/aztec-packages/pull/1919/commits/1159e07f3444c221c6be534a76a5e601eae9e741